src/blockencodings.cpp

```diff
@@ -167,7 +167,9 @@ ReadStatus PartiallyDownloadedBlock::InitData(const CBlockHeaderAndShortTxIDs& c
             break;
     }
 
-    LogDebug(BCLog::CMPCTBLOCK, "Initialized PartiallyDownloadedBlock for block %s using a cmpctblock of size %lu\n", cmpctblock.header.GetHash().ToString(), GetSerializeSize(cmpctblock));
+    if (LogAcceptCategory(BCLog::CMPCTBLOCK, BCLog::Level::Debug)) {
+        LogDebug(BCLog::CMPCTBLOCK, "Initialized PartiallyDownloadedBlock for block %s using a cmpctblock of size %lu\n", cmpctblock.header.GetHash().ToString(), GetSerializeSize(cmpctblock));
+    }
```

Comment:
> // Use a macro instead of a function for conditional logging to prevent
> // evaluating arguments when logging for the category is not enabled.

Thanks, dropped the logger related commit. What's the reason for the logger avoiding evaluation but `Assume` not doing it?

> What is the goal? Is this an optimization?

As stated in the description `a costly calculation, we can deduplicate it`. I'll clarify further in the description.

> If yes, how can it be observed?

A quick peek can be seen in https://github.com/bitcoin-dev-tools/benchcoin/pull/66, but I'm currently running several full reindex-chainstates to see how much effect it has - previous measurements show 2-3%.